testing/deprecated_test.py

```diff
@@ -51,6 +51,52 @@ def test_fillfixtures_is_deprecated() -> None:
         _pytest.fixtures.fillfixtures(mock.Mock())
 
 
+def test_hooksec_marks_are_deprecated():
+    from _pytest.config import PytestPluginManager
+
+    pm = PytestPluginManager()
+
+    class DeprecatedHookMarkerSpec:
+        def pytest_bad_hook(self):
+            pass
+
+        pytest_bad_hook.historic = True  # type: ignore[attr-defined]
+
+    with pytest.warns(
+        PytestDeprecationWarning, match="instead of pytest.mark"
+    ) as recorder:
+        pm.add_hookspecs(DeprecatedHookMarkerSpec)
+    (record,) = recorder
+    assert (
+        record.lineno
+        == DeprecatedHookMarkerSpec.pytest_bad_hook.__code__.co_firstlineno
+    )
+    assert record.filename == __file__
+
+
+def test_hookimpl_marks_are_deprecated():
```

Comment:
This name is a bit misleading, you are testing setting an attribute, not a `mark`.

While we could rename it, I think it is better to change the test to use a `@pytest.mark.tryfirst` in the hook, as otherwise we don't have a test to check the deprecation is issued in that case.